core/src/gui/widgets/combo_as_slider.cpp

```diff
@@ -0,0 +1,30 @@
+#include <gui/widgets/combo_as_slider.h>
+#include <imgui.h>
+#include <imgui_internal.h>
+
+
+namespace ImGui {
+    bool ComboAsSlider(const char* label, int* current_item, const char* items_separated_by_zeros){
+        int items_count = 0;
+        const char* p = items_separated_by_zeros;
+        while (*p)
+        {
+            p += strlen(p) + 1;
+            items_count++;
+        }
+        char** items_starts = new char*[items_count];
```

Comment:
Heap allocation and deallocation on every frame is VERY BAD for performance. Also code style from contributing.md isn't respected.

Reply:
I rewrote this part, please take a look.